kernel/src/expressions/scalars.rs

```diff
@@ -847,6 +848,13 @@ mod tests {
         expect_fail_parse("0.E170141183460469231731687303715884105727", 1, 0);
     }
 
+    #[test]
+    fn test_variant_expect_fail() {
+        let s = PrimitiveType::Variant;
+        let res = s.parse_scalar("1");
+        assert!(res.is_err(), "Variant scalar works even though it is not supposed to.");
```

Comment:
let's check the actual error (could `assert!(matches!(...))` or something similar)